src/test/kotlin/com/atkinsondev/opentelemetry/build/OpenTelemetryBuildPluginSpansIntegrationTest.kt

```diff
@@ -0,0 +1,188 @@
+
+        expectThat(buildResult.task(":test")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+
+        // Parse trace ID from build output
+        val traceId = Regex("OpenTelemetry build trace ID (\\w+)").find(buildResult.output)!!.groupValues[1]
+
+        // Fetch trace data from Jaeger
+        val httpClient = OkHttpClient.Builder().build()
+        val request =
+            Request.Builder()
+                .url("http://localhost:${jaegerContainer.getMappedPort(JAEGER_QUERY_PORT)}/api/traces/$traceId")
+                .get()
+                .build()
+        val resp = httpClient.newCall(request).execute()
+        expectThat(resp.code).isEqualTo(200)
+
+        val decoded = json.decodeFromString(JaegerApiResponse.serializer(), resp.body!!.string())
+        // Lazy as the order isn't always correct (e.g. the result returned by Jaeger returns a child span before the parent)
+        val depths = mutableMapOf<String, () -> Int>()
+        val orderedSpans =
+            // Even though so far we only expect `data` to contain a single element, a flatMap should make sense here
+            decoded.data.flatMap { d ->
+                d.spans.map { span ->
+                    val depth =
+                        if (span.references.isEmpty()) {
+                            {
+                                0
+                            }
+                        } else {
+                            val parentSpanId =
+                                span.references
+                                    .also {
+                                        assert(it.size == 1)
+                                    }
+                                    .first()
+                                    .spanID
+                            {
+                                depths[parentSpanId]!!() + 1
+                            }
+                        }
+                    depths[span.spanID] = depth
+                    FlatSpan(
+                        operationName = span.operationName,
+                        startTime = span.startTime,
+                        depth = depth,
+                    )
+                }
+            }.sortedBy {
+                it.startTime
+            }
+        val orderedSpansNamesWithDepth =
+            orderedSpans.map {
+                ">".repeat(it.depth()) + " ${it.operationName}"
+            }
+
+        // Use assertLinesMatch, as it has nice support for regexes
+        assertLinesMatch(
+            listOf(
+                " junit\\d+-build",
+                "> :compileKotlin",
+                "> :processResources",
+                "> :processTestResources",
+                "> :compileJava",
+                "> :classes",
+                "> :compileTestKotlin",
+                "> :compileTestJava",
+                "> :testClasses",
+                "> :test",
+                ">> FooTest foo should return bar()",
+            ),
+            orderedSpansNamesWithDepth,
```

Comment:
I think the end result of having the generated spans tested here so clearly is pretty cool, right!

The code above to process the result of the Jaeger Api could probably be improved, made more clear but it works 😃